doc/reference/guidelines.rst

```diff
@@ -0,0 +1,48 @@
+.. highlight:: python
+
+=================
+Coding Guidelines
+=================
+
+Python
+======
```

Comment:
> we ask to R&D India to sort their imports in this order 1. python standard lib 2. other libs 3. internal odoo imports

And we should mandate absolute imports (so in-package relative imports should always be in the form `from ..foo import bar`)